lib/modules/commentHidePersistor.js

```diff
@@ -16,115 +16,66 @@ module.include = [
-	await hideHiddenThings();
-};
-
-let allHiddenThings = {};
-let hiddenKeys = [];
-let hiddenThings = [];
-const hiddenThingsKey = location.pathname;
-const maxKeys = 100;
-const pruneKeysTo = 50;
-
-function bindToHideLinks() {
-	/**
-	 * For every expand/collapse link, add a click listener that will
-	 * store or remove the comment ID from our list of hidden comments.
-	 **/
-	$('body').on('click', 'a.expand', function() {
-		const $thing = $(this).parents('.thing:eq(0)');
-		const thingId = $thing.attr('data-fullname');
-		const collapsing = $thing.is('.collapsed') || $(this).parent().parent().hasClass('noncollapsed');
-
-		/* Add our key to pages interacted with, for potential pruning later */
-		if (!hiddenKeys.includes(hiddenThingsKey)) {
-			hiddenKeys.push(hiddenThingsKey);
-		}
-
-		if (collapsing) {
-			addHiddenThing(thingId);
-		} else {
-			removeHiddenThing(thingId);
-		}
-	});
-}
-
-async function loadHiddenThings() {
-	const hidePersistorData = await hidePersistorStorage.get();
-
-	if (hidePersistorData) {
-		allHiddenThings = hidePersistorData.hiddenThings;
-		hiddenKeys = hidePersistorData.hiddenKeys;
+module.beforeLoad = async () => {
+	if (!currentId) throw new Error('Could not find comment page id');
 
-		/**
-		 * Prune allHiddenThings of old content so it doesn't get
-		 * huge.
-		 **/
-		if (hiddenKeys.length > maxKeys) {
-			const pruneStart = maxKeys - pruneKeysTo;
-			const newHiddenThings = {};
-			/* Recreate our object as a subset of the original */
-			const newHiddenKeys = hiddenKeys.slice(pruneStart);
-
-			for (const hiddenKey of newHiddenKeys) {
-				newHiddenThings[hiddenKey] = allHiddenThings[hiddenKey];
-			}
+	await restoreCommentCollapse();
+};
 
-			allHiddenThings = newHiddenThings;
-			hiddenKeys = newHiddenKeys;
-			syncHiddenThings();
-		}
+module.go = () => {
+	listenToCommentCollapse();
+};
 
-		if (allHiddenThings[hiddenThingsKey]) {
-			hiddenThings = allHiddenThings[hiddenThingsKey];
-		}
+module.afterLoad = async () => {
+	// Clean counts every six hours
+	const lastClean = await lastCleanStorage.get() || 0;
+	if ((Date.now() - lastClean) > 0.25 * DAY) {
+		pruneOldEntries();
 	}
-}
+};
 
-function addHiddenThing(thingId) {
-	if (!hiddenThings.includes(thingId)) {
-		hiddenThings.push(thingId);
+async function pruneOldEntries() {
+	const now = Date.now();
+	lastCleanStorage.set(now);
+	const keepTrackPeriod = DAY * parseInt(expireAfterDays, 10);
+	for (const [id, data] of Object.entries(await entryStorage.getAll())) {
+		if ((now - data.updateTime) > keepTrackPeriod) {
+			entryStorage.delete(id);
+		}
 	}
-	syncHiddenThings();
 }
 
-function removeHiddenThing(thingId) {
-	const i = hiddenThings.indexOf(thingId);
-	if (i !== -1) {
-		hiddenThings.splice(i, 1);
+async function restoreCommentCollapse() {
+	const { collapsedThings } = await entryStorage.get(currentId);
+	if (collapsedThings) {
+		watchForThings(['comment'], thing => {
+			if (Reflect.has(collapsedThings, thing.getFullname())) {
```

Comment:
FWIW you can use `in` here if you want, I don't really care about enforcing the use of `Reflect` (mainly enforce `Reflect.apply` just so it stands out more than `fn.call`). If you prefer it this way though then feel free to keep it.